### PR TITLE
Set up collections page and refactored code

### PR DESCRIPTION
**Pull request by redliz5808**

### CODE REVIEW

File: src/components/CollectionPhotos/CollectionPhotos.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import axios from "axios";
+import { Icon } from "components";
+import { FaHeart } from "react-icons/fa";
+import { Container, StyledLink, StyledDiv } from "./collectionPhotos.styles";
+
+class CollectionPhotos extends React.Component {
+  state = {
+    data: null,
+  };
```
Comment (aOrro) [inline src/components/CollectionPhotos/CollectionPhotos.js:10]: loading flag?

File: src/components/SearchCollections/SearchCollections.js [hunk 2]
```diff
@@ -62,7 +62,10 @@ class SearchCollections extends React.Component {
           <StyledDiv>
             {collectionData.results.map((collection) => {
```
Comment (aOrro) [inline src/components/SearchCollections/SearchCollections.js:62]: I guess above you could be doing just one check and get rid of the other, then just do a ternary and render both options depending if true or false

File: src/components/SearchPhotos/SearchPhotos.js [hunk 3]
```diff
@@ -30,19 +30,18 @@ class SearchPhotos extends React.Component {
...
       this.getPhotoData(searchTerm);
-      this.setState({ searchTerm });
     }
   }
 
   render() {
-    const { photoData, searchTerm } = this.state;
+    const { searchTerm } = this.props;
+    const { photoData } = this.state;
     const readyWithoutPhotos = photoData && photoData.total === 0;
```
Comment (aOrro) [inline src/components/SearchPhotos/SearchPhotos.js:44]: same with this? better a ternary with one check?

File: src/components/UserStats/UserStats.js [hunk 4]
```diff
@@ -12,14 +12,14 @@ class UserStats extends React.Component {
 
   loadingBar = React.createRef();
 
-  baseUrl = `${process.env.REACT_APP_API_BASE_URL}/users`;
-
   retrieveStats = async (username) => {
+    const baseUrl = `${process.env.REACT_APP_API_BASE_URL}/users`;
```
Comment (aOrro) [inline src/components/UserStats/UserStats.js]: I kinda find this useless, you already have a base url, why creating a new one by just adding a slash/whatever? why not doing the whole thing straight on the api request? Again this is a matter of personal preference, theres nothing wrong with it

File: src/pages/Collection/Collection.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        {readyToLoad && (
+          <>
+            <h1>{data.title}</h1>
+            {data.description ? <h3>{data.description}</h3> : null}
+            <StyledLink to={`/user/${data.user.username}`}>
+              {data.user.name}
+            </StyledLink>
+            <Tags>
+              {tagsAvailable &&
+                data.tags.map((tag, index) => {
```
Comment (aOrro) [inline src/pages/Collection/Collection.js:52]: nice one

File: src/pages/User/User.js [hunk 6]
```diff
@@ -44,27 +44,28 @@ class User extends React.Component {
...
     return (
       <>
         <LoadingBar color="#6958f2" ref={this.loadingBar} />
-        {this.state.data && !this.state.isLoading && (
+        {readyToLoad && (
           <>
             <img src={data.profile_image.large} alt={data.name} />
             <h1>{data.name}</h1>
-            {data.badge ? <Verified>Verified ✓</Verified> : ""}
+            {data.badge ? <Verified>Verified ✓</Verified> : null}
```
Comment (aOrro) [inline src/pages/User/User.js]: data.badge && <Verified>Verified ✓</Verified> ? I dont think there's need for a ternary here...

File: src/pages/User/User.js [hunk 7]
```diff
@@ -44,27 +44,28 @@ class User extends React.Component {
...
-            {data.instagram_username && (
+            {data.instagram_username ? (
               <InstagramUser
                 href={`https://www.instagram.com/${data.instagram_username}`}
                 target="_blank"
               >
                 @{data.instagram_username}
               </InstagramUser>
-            )}
+            ) : null}
```
Comment (aOrro) [inline src/pages/User/User.js]: Why this change? same as before... If it is because of something Cristian said, dont listen to me but again i dont need theres any need of a ternary here

**Review by aOrro — COMMENTED** [review]
nice one! keep it up!!

File: src/components/UserStats/UserStats.js (hunk 4 above)
Comment (yoloOnTheBattlefield) [reply src/components/UserStats/UserStats.js]: this is ok in my opinion, it's not bad or good, it's just an opinion, the only bad thing is if you are not consistent with it

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]
https://www.loom.com/share/3ad4ae625b6a4395861151c791c2a5ba

File: src/components/UserStats/UserStats.js (hunk 4 above)
Comment (redliz5808) [reply src/components/UserStats/UserStats.js]: I tried it out and I do think I like it better putting the whole thing in the api request. I originally had it set up like this because of doing multiple api requests in one file, but now I've changed that so it makes more sense to do it the other way. Thanks!

File: src/components/SearchCollections/SearchCollections.js (hunk 2 above)
Comment (redliz5808) [reply src/components/SearchCollections/SearchCollections.js:62]: If I use a ternary, while the information is loading it will display the message that there is no data for collections (or photos or users depending on which tab you're on) so that won't really work the same or the way I want it to.
